### ISSUE TITLE
OpenAI v1 Compatible REST implementation for models and completions

### ISSUE THREAD

**Issue by Fmstrat** [imartinez/privateGPT#577]
Many of the other tickets are generic surrounding API implementations. This is a specific enhancement surrounding OpenAI compatibility for use with existing plugins like OnlyOffice (https://github.com/ONLYOFFICE/onlyoffice.github.io).

**Describe the solution you'd like**
Implement the primary 2 use cases (models and completions) for external software integrations. Models should be fairly straight forward as it would just return whatever is configured in the `env`, completions seem like it wouldn't be too far off from prompting.

In other words, add API endpoints that resolve as:
- `/v1/models`: https://platform.openai.com/docs/api-reference/models
- `/v1/completions`: https://platform.openai.com/docs/api-reference/completions

**Describe alternatives you've considered**
Sending all the datas to OpenAI.

Note: Depending on response from @imartinez or other participating devs, I may be interested in taking this on.

**Comment by Fmstrat** [comment]
@imartinez did this complete or close? Was there a PR? Thanks!

**Comment by imartinez** [comment]
> @imartinez did this complete or close? Was there a PR? Thanks!

Completions API was created following the exact input and output of OpenAIs completions API. What changes would you add?

**Comment by Fmstrat** [comment]
None at all, I just noticed it closed with no PR linked. Thank you!
